app/workers/functions.php

```diff
@@ -143,242 +345,54 @@ function: $function,
                  *      If error count bigger than allowed change status to pause
                  */
 
-                // Reschedule
-                $function = $database->getDocument('functions', $function->getId());
-
-                if (empty($function->getId())) {
-                    throw new Exception('Function not found (' . $function->getId() . ')');
-                }
-
-                if ($functionOriginal->getAttribute('schedule') !== $function->getAttribute('schedule')) { // Schedule has changed from previous run, ignore this run.
-                    return;
-                }
-
-                if ($functionOriginal->getAttribute('scheduleUpdatedAt') !== $function->getAttribute('scheduleUpdatedAt')) { // Double execution due to rapid cron changes, ignore this run.
-                    return;
-                }
-
-                $cron = new CronExpression($function->getAttribute('schedule'));
-                $next = DateTime::format($cron->getNextRunDate());
-
-                $function = $function
-                    ->setAttribute('scheduleNext', $next)
-                    ->setAttribute('schedulePrevious', DateTime::now());
-
-                $function = $database->updateDocument(
-                    'functions',
-                    $function->getId(),
-                    $function
-                );
-
-                $reschedule = new Func();
-                $reschedule
-                    ->setFunction($function)
-                    ->setType('schedule')
-                    ->setUser($user)
-                    ->setProject($project)
-                    ->schedule(new \DateTime($next));
-                ;
-
-                $this->execute(
-                    project: $project,
-                    function: $function,
-                    dbForProject: $database,
-                    trigger: 'schedule'
-                );
-
+                call_user_func($execute, $project, $function, $dbForProject, 'schedule', null, null, null, null, null, null);
                 break;
         }
-    }
+    });
 
-    private function execute(
-        Document $project,
-        Document $function,
-        Database $dbForProject,
-        string $trigger,
-        string $executionId = null,
-        string $event = null,
-        string $eventData = null,
-        string $data = null,
-        ?Document $user = null,
-        string $jwt = null
-    ) {
-
-        $user ??= new Document();
-        $functionId = $function->getId();
-        $deploymentId = $function->getAttribute('deployment', '');
-
-        /** Check if deployment exists */
-        $deployment = $dbForProject->getDocument('deployments', $deploymentId);
-
-        if ($deployment->getAttribute('resourceId') !== $functionId) {
-            throw new Exception('Deployment not found. Create deployment before trying to execute a function', 404);
-        }
+$server
```

Comment:
We can define both: 
```php
$adapter  = new Queue\Adapter\Swoole($client, $workerNumber, Event::FUNCTIONS_QUEUE_NAME);
$server   = new Queue\Server($adapter);
```

Inside worker.php and then set the logger in the same file and use it as a global logger for all workers.

Reply:
Problem, we use a different queue name for each worker

Reply:
Moved server setup and logger into worker.php